.github/tool_checkmake.sh

```diff
@@ -0,0 +1,143 @@
+#    SERVICES, HOWEVER CAUSED, REGARDLESS OF THE THEORY OF LIABILITY (CONTRACT,
+#    TORT OR OTHERWISE) AND EVEN IF THE AUTHOR HAS BEEN ADVISED OF THE
+#    POSSIBILITY OF SUCH DAMAGES. SOME JURISDICTIONS DO NOT ALLOW THE EXCLUSION
+#    OR LIMITATION OF LIABILITY FOR PERSONAL INJURY, OR OF INCIDENTAL OR
+#    CONSEQUENTIAL DAMAGES, SO THIS LIMITATION MAY NOT APPLY TO YOU. In no event
+#    shall THE AUTHOR's total liability to you for all damages (other than as may
+#    be required by applicable law in cases involving personal injury) exceed
+#    the amount of five dollars ($5.00). The foregoing limitations will apply
+#    even if the above stated remedy fails of its essential purpose.
+################################################################################
+#
+# .github/tool_checkmake.sh
+readonly SCRIPT_NAME="${0##*/}"
+
+# local build path fix-up
+if [[ -d "./checkmake" ]] && [[ ":$PATH:" != *":./checkmake:"* ]] ; then
+	PATH="${PATH:+"$PATH:"}./checkmake" ;
+	export PATH ;
+fi
+
+# USAGE:
+#  ~$ check_command CMD
+# Arguments:
+# CMD (Required) -- Name of the command to check
+# Results:
+#    exits 64 -- missing required argument
+#    exits 126 -- check complete and has failed, can not find given command.
+#    returns successful -- check complete and command found to be executable.
+function check_command() {
+	test -z "$1" && { printf "%s\n" "::warning file=${SCRIPT_NAME},title=BUG::Command name is required to check for existence." >&2 ; exit 64 ; } ;
+	local cmd="$1" ;
+	test -x "$(command -v ${cmd})" || { printf "%s\n" "::error file=${SCRIPT_NAME},title=MISSING::Required command '${cmd}' is not found." >&2 ; exit 126 ; } ;
+}  # end check_command()
+# propagate/export function to sub-shells
+export -f check_command
+
+# Check required commands
+check_command sed ;
+check_command grep ;
+check_command cut ;
+check_command go ;
+
+check_command checkmake ;
+
+# USAGE:
+#   ~$ usage
+# Arguments:
+#   None
+# Results:
+#   Prints usage information and exits with status code 2.
+function usage() {
+	printf "Usage: %s <makefile_path>\n" "${SCRIPT_NAME}" >&2
+	exit 2
+}
+
+# Validate parameters
+if [[ ( "$#" -lt 1 ) ]]; then
+	usage
+fi
+readonly FILE="${1}"
+readonly EMSG="Checkmake linter complained."
+
+# Check if file exists
+if [[ ! ( -f "${FILE}" ) ]]; then
+	printf "%s\n" "::error file=${FILE},title=MISSING::File '${FILE}' not found." >&2
+	exit 64
+fi
+
+# Main functionality
+process_checkmake_output() {
+	local file="$1"
+	local emsg="$2"
+
+	if ! output=$(checkmake "${file}" 2>&1); then
+		printf "%s\n" "::error::checkmake failed!"
+
+		printf "%s\n" "${output}" | \
+			sed -e 's/   /:/g' | \
+			tr -s ':' | \
+			cut -d: -f 3-5 | \
+			grep -F "${file}" | \
+			sed -E -e 's/^[[:space:]]+//g' | \
+			while IFS= read -r line; do
+				printf "%s\n" "::warning file=${file},title=LINT::${line} ${emsg}" >&2
+			done
+		return 1
+	else
+		printf "%s\n" "::notice file=${file},title=LINT::OK - No lint errors." >&2
+	fi ;
+}
```

Comment:
🙅🏻 current implementation works, your suggestion only adds frivolous error handling, and breaks the linting errors by hiding them behind an early error based on the exit of the tool rather than handling the errors from the tool. The current implementation in this specific case is better, perhaps we can re-visit improvements the next time we need to update this particular file.